hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -917,6 +917,7 @@ private static Object rewritePrimaryType(Object oldValue, Schema oldSchema, Sche
         case DOUBLE:
         case BYTES:
         case STRING:
+        case ENUM:
```

Comment:
Can we move this to the caller of `rewritePrimaryType`? This method was only meant for primitive types. I don't know why even `FIXED` is here.

Reply:
Makes sense.  Moved to the caller `rewriteRecordWithNewSchemaInternal()`.